This patch is based on the commit described below, taken from the upstream
time64 pull request and modified for use in the Debian package by Peter
Michael Green.

commit 4385156cfbdfd908881a606e9db5d68a0555847c
Author: Ola x Nilsson <olani@axis.com>
Date:   Mon Mar 20 14:29:55 2023 +0100

    gnu: Update struct semid_ds for 64-bit time

Index: libc/src/unix/linux_like/linux/gnu/b32/mod.rs
===================================================================
--- libc.orig/src/unix/linux_like/linux/gnu/b32/mod.rs
+++ libc/src/unix/linux_like/linux/gnu/b32/mod.rs
@@ -109,25 +109,36 @@ s! {
 
     pub struct semid_ds {
         pub sem_perm: ipc_perm,
-        #[cfg(target_arch = "powerpc")]
+        #[cfg(all(not(gnu_time64_abi), target_arch = "powerpc"))]
         __reserved: crate::__syscall_ulong_t,
         pub sem_otime: crate::time_t,
         #[cfg(not(any(
+            gnu_time64_abi,
             target_arch = "mips",
             target_arch = "mips32r6",
             target_arch = "powerpc"
         )))]
         __reserved: crate::__syscall_ulong_t,
-        #[cfg(target_arch = "powerpc")]
+        #[cfg(all(not(gnu_time64_abi), target_arch = "powerpc"))]
         __reserved2: crate::__syscall_ulong_t,
         pub sem_ctime: crate::time_t,
         #[cfg(not(any(
+            gnu_time64_abi,
             target_arch = "mips",
             target_arch = "mips32r6",
             target_arch = "powerpc"
         )))]
         __reserved2: crate::__syscall_ulong_t,
         pub sem_nsems: crate::__syscall_ulong_t,
+        #[cfg(all(gnu_time64_abi,
+                  not(any(
+                      target_arch = "mips",
+                      target_arch = "mips32r6",
+                      target_arch = "powerpc",
+                      target_arch = "arm",
+                      target_arch = "x86"
+                  ))))]
+        __reserved2: crate::__syscall_ulong_t,
         __glibc_reserved3: crate::__syscall_ulong_t,
         __glibc_reserved4: crate::__syscall_ulong_t,
     }
